src/nupic/engine/RegionImplFactory.cpp

```diff
@@ -59,6 +59,12 @@ namespace nupic
   // Allows the user to add custom regions
   void RegionImplFactory::registerPyRegion(const std::string module, const std::string className)
   {
+    // Verify that no regions exist with the same className
+    for (auto pyr=pyRegions.begin(); pyr!=pyRegions.end(); pyr++)
```

Comment:
Can you use the `for (auto pyRegion : pyRegions)` form?

If not, make sure to put spaces before and after the `=` and `!=`